customer-service/src/main/scala/domain/services/CustomerDetailsService.scala

```diff
@@ -1,9 +1,12 @@
 package domain.services
 
 import cats.data.OptionT
-import domain.{CustomerDetails, CustomerId}
+import domain.CustomerDetails
 import domain.repositories.CustomerDetailsRepository
 
-class CustomerDetailsService[F[_]](repository: CustomerDetailsRepository[F]) {
+class CustomerDetailsService[F[_]](executionPriority: Int, repository: CustomerDetailsRepository[F]) {
```

Comment:
it is just another trap :)

remember that: 1 domain concept = 1 service
so single service in general should have 1 repository and multiple other services, however notice that `CustomerDetailsService` there is no difference between details from legacy system and that new v2 microservice

so the appropriate design would be to have two repositories there, as you should never have two different **services** for one **domain** concept, you should not have two different **repositories** for the same **data access** protocol, but it is perfectly OK to have two different repositories for two different data access protocols, example: serviceX provides two data access layers - one is GRPC - very fast but very small (only small portion of fields available) and full HTTP endpoint that is really slow and gives big payload; depend on granularity of data ur needing for processing u will call grpc or http, however - it should be two repositories under one domain service

that is clear?